smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -820,6 +944,124 @@ private void readHeaders(
         return payloadBindings;
     }
 
+    private void generateEventStreamDeserializer(GenerationContext context, MemberShape member, Shape target) {
+        TypeScriptWriter writer = context.getWriter();
+        writer.openBlock("context.eventStreamMarshaller.deserialize(", ");", () -> {
+            writer.write("output.body,");
+            writer.openBlock("async event => {", "}", () -> {
+                writer.write("const eventName = Object.keys(event)[0];");
+                writer.write("const eventValue = event[eventName];");
+                writer.openBlock("const eventHeader = Object.entries(eventValue.headers).reduce(",
```

Comment:
This should probably be `eventHeaders` for clarity, even if it's reduced to a single string.